hazelcast/src/main/java/com/hazelcast/config/ExternalDataStoreConfig.java

```diff
@@ -174,4 +189,40 @@ public String toString() {
                 + ", properties=" + properties
                 + '}';
     }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeString(name);
+        out.writeString(className);
+        out.writeBoolean(shared);
+        out.writeInt(properties.size());
+        for (String key : properties.stringPropertyNames()) {
+            out.writeString(key);
+            out.writeString(properties.getProperty(key));
+        }
+    }
+
+    @Override
+    public void readData(ObjectDataInput in) throws IOException {
+        name = in.readString();
+        className = in.readString();
+        shared = in.readBoolean();
+        properties.clear();
```

Comment:
Minor: This seems unnecessary to me. We call readData after creating the instance with the empty constructor, and there, the properties is already empty

Reply:
fixed: https://github.com/hazelcast/hazelcast/pull/21930/commits/afd92595fb8d447aee75088309b75672fd8299ea